scripts/wise.ts

```diff
@@ -45,7 +50,75 @@ program.command('check-expense <expenseId>').action(async expenseId => {
+  console.dir({ transfer, recipient, quote, paymentOption }, { depth: null });
+
+  if (
+    'price' in expense.data.paymentOption && // Check if it is QuoteV3
+    paymentOption.price.priceDecisionReferenceId !== expense.data.paymentOption.price?.priceDecisionReferenceId
+  ) {
+    console.warn(
+      `Payment option mismatch! Reference stored in DB: ${expense.data.paymentOption.price?.priceDecisionReferenceId}, reference from Wise: ${paymentOption.price.priceDecisionReferenceId}`,
+    );
+  }
+  sequelize.close();
+});
+
+const checkPaymentProcessorFee = ({ expense, quote }) => {
+  const paymentOption = quote.paymentOptions.find(p => p.payIn === 'BALANCE' && p.payOut === quote.payOut);
+  assert.equal(expense.data.paymentOption.fee.total, paymentOption.fee.total, `Fee mismatch`);
+};
+
+program.command('check-host <hostSlug> [since]').action(async (hostSlug, since) => {
+  since = since || moment.utc().startOf('year');
+  console.log(`Checking expense for host ${hostSlug} since ${since}`);
+  const host = await models.Collective.findOne({ where: { slug: hostSlug } });
+  const [connectedAccount] = await host.getConnectedAccounts({
+    where: { service: Service.TRANSFERWISE, deletedAt: null },
+  });
+  if (!connectedAccount) {
+    return printAndExit(`${host.slug} not connected to Wise`, 'error');
+  }
+  const expenses = await models.Expense.findAll({
+    where: {
+      HostCollectiveId: host.id,
+      status: 'PAID',
+      updatedAt: { [Op.gte]: since },
+      data: { transfer: { [Op.ne]: null } },
+      feesPayer: 'COLLECTIVE',
+    },
+    include: [
+      { model: models.Collective, as: 'collective' },
+      { model: models.Collective, as: 'host', required: true },
+      {
+        model: models.PayoutMethod,
+        as: 'PayoutMethod',
+        required: true,
+        where: { type: PayoutMethodTypes.BANK_ACCOUNT },
+      },
+    ],
+  });
+
+  const problems = [['ExpenseId', 'CollectiveSlug', 'CreatedAt', 'Problem', 'Actual', 'Expected']];
+  console.log(`Found ${expenses.length} expenses...`);
+  for (const expense of expenses) {
+    const transfer = await transferwiseLib.getTransfer(connectedAccount, expense.data.transfer['id']);
+    const quote = await transferwiseLib.getQuote(connectedAccount, transfer.quoteUuid);
+    try {
+      checkPaymentProcessorFee({ expense, quote });
+      console.log(`✅ Expense #${expense.id} for ${expense.collective.slug}`);
+    } catch (e) {
+      problems.push([expense.id, expense.collective.slug, expense.createdAt, e.message, e.actual, e.expected]);
+      console.log(`❌ Expense #${expense.id} for ${expense.collective.slug} failed check: ${e.message}`);
+    }
+  }
+
+  if (problems.length > 1) {
+    console.error(`Found ${problems.length - 1} problems, writting it to wise-check.csv`);
+    fs.writeFileSync('wise-check.csv', problems.map(row => row.join(',')).join('\n'));
```

Comment:
I suppose we'll follow up with a script to fix transactions for those past entries, based on the CSV results?

Reply:
Yes, this should be addressed in the coming weeks together with affected fiscal hosts.